### PR TITLE
[Feature/pur 610 alert]: Alert component

### PR DESCRIPTION
**Pull request by kevin-ho87**
**Ticket:**
https://aesoponline.atlassian.net/browse/PUR-610

**Task:**
Create success Alert component
image

**Dev description:**
- Alert component
- Props
  - `severity` - renders tick icon
  - `copy` - text
- Add tick icon to `<Icon />` 

**References:**
- https://w3c.github.io/aria-practices/examples/alert/alert.html (accessibility)
- https://mui.com/components/alert/ (structure)

### CODE REVIEW

**Review by cyrusxzw — APPROVED** [review]

File: src/components/Alert/Alert.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Alert } from './Alert';
+import { AlertFixture } from './Alert.fixture';
+
+describe('<Alert />', () => {
+  it('should render snapshot', () => {
```
Comment (MkMan) [inline src/components/Alert/Alert.spec.tsx:7]: Snapshot tests can be burdensome to maintain and they don't test something specific. I liked the views shared in this [tweet](https://twitter.com/searls/status/919594505938112512) and [article](https://kentcdodds.com/blog/effective-snapshot-testing).

I think more specific tests more valuable. Happy to discuss 🙂

**Review by MkMan — COMMENTED** [review]
Don't forget to export the component from GEL 😉

File: src/components/Alert/Alert.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import cx from 'classnames';
+import { useThemeContext } from '~/contexts';
+import type { AlertType } from './Alert.types';
+import { Icon } from '~/components/Icon';
+import styles from './Alert.module.css';
+
+const Alert: AlertType = ({ className, theme, copy, severity }) => {
+  const currentTheme = useThemeContext(theme, 'dark');
```
[the remark this answers is not in the document]
Comment (kevin-ho87) [reply src/components/Alert/Alert.tsx:9]: do you have an example?

File: src/components/Alert/Alert.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    const { container } = render(
+      <Alert copy={AlertFixture.copy} severity="success" />,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should not render copy', () => {
```
[the remark this answers is not in the document]
Comment (kevin-ho87) [reply src/components/Alert/Alert.spec.tsx:14]: I am testing on if the `<p>` tag should render depending on `copy` prop

File: src/components/Alert/Alert.spec.tsx (hunk 1 above)
Comment (kevin-ho87) [reply src/components/Alert/Alert.spec.tsx:7]: That makes sense, in testing the differences. I only added snapshot as it was the norm.

File: src/components/Alert/Alert.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (MkMan) [reply src/components/Alert/Alert.tsx:9]: ### When it's set in destructured props

https://github.com/aesop/aesop-gel/blob/21b0803f8c993acec17236c62699f7916b207c36/src/components/Checkbox/Checkbox.tsx#L20

https://aesop.github.io/aesop-gel/?path=/docs/form-elements-checkbox--base-component

Screen Shot 2021-11-16 at 3 32 34 pm

### When it's **not** set in destructured props

https://github.com/aesop/aesop-gel/blob/21b0803f8c993acec17236c62699f7916b207c36/src/components/Button/Button.tsx#L23

https://aesop.github.io/aesop-gel/?path=/docs/components-button--base-component

image

**Comment by kevin-ho87** [comment]
> Don't forget to export the component from GEL 😉

ah yes, thanks for that 👍

File: src/components/Alert/Alert.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevin-ho87) [reply src/components/Alert/Alert.tsx:9]: got it.

File: src/components/Alert/Alert.spec.tsx (hunk 1 above)
Comment (kevin-ho87) [reply src/components/Alert/Alert.spec.tsx]: updated test

**Review by MkMan — APPROVED** [review]

**Pull request merged**